### PR TITLE
WT-8853 Update import_compatibility_test with CMake

### PR DESCRIPTION
**Pull request by etienneptl**
This commit includes the different changes from https://github.com/wiredtiger/wiredtiger/pull/7317 related to `import_compatibility_test.sh`. No additional changes were made, the exact same changes were cherry-picked.

### CODE REVIEW

**Review by korteland — APPROVED** [review]
Looks good to me, the indentation in the if/else when we're setting the `config` string is a bit off but that's in the original PR.

File: test/evergreen/import_compatibility_test.sh [hunk 1]
```diff
@@ -21,11 +21,21 @@ build_branch()
 
     git checkout --quiet "$1"
 
-    config=""
-    config+="--enable-snappy "
-    config+="--disable-standalone-build "
-    (sh build_posix/reconf &&
-         ./configure $config && make -j $(grep -c ^processor /proc/cpuinfo)) > /dev/null
+    if [ "${build_sys[$1]}" == "cmake" ]; then
```
Comment (ajmorton) [inline test/evergreen/import_compatibility_test.sh:24]: Do we also need this fix in `compatibility_test_for_releases.sh`? [Example from original PR](https://github.com/wiredtiger/wiredtiger/pull/7317/files#diff-db594fea5783b6224edb60d31c47b6b25bec3646a7f49eedaeb2388ecb82646cR47).

The current code in mongo-5.0 [still references build_posix](https://github.com/wiredtiger/wiredtiger/blob/3f3a64194497f5a753ba64417ae717e76b6e15f5/test/evergreen/compatibility_test_for_releases.sh#L51).

Comment (etienneptl) [reply test/evergreen/import_compatibility_test.sh:24]: Please see #7587

**Review by ajmorton — APPROVED** [review]
LGTM

**Comment by etienneptl** [comment]
Closing at this will be addressed as part of [WT-8874](https://jira.mongodb.org/browse/WT-8874).

**Pull request closed without merging**